framework/src/modules/token/method.ts

```diff
@@ -689,15 +693,47 @@ export class TokenMethod extends BaseMethod {
 		methodContext: MethodContext,
 		chainID: Buffer,
 	): Promise<void> {
-		await this.stores.get(SupportedTokensStore).supportChain(methodContext, chainID);
+		const allSupportedTokenEntries = await this._isAllTokensSupported(methodContext);
+
+		if (allSupportedTokenEntries) {
+			return;
+		}
+
+		if (chainID.equals(this._config.ownChainID)) {
+			return;
+		}
+
+		const supportedTokens = await this._getSupportedTokens(methodContext, chainID);
+
+		if (supportedTokens) {
+			await this.stores
+				.get(SupportedTokensStore)
+				.set(methodContext, chainID, { supportedTokenIDs: [] });
+		}
```

Comment:
```
await this.stores
				.get(SupportedTokensStore)
				.set(methodContext, chainID, { supportedTokenIDs: [] });
```
Regardless of pre-existing data or not, i think we can always overwrite with this